transforms/universal/web2parquet/dpk_web2parquet/transform.py

```diff
@@ -0,0 +1,137 @@
+        """
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of NOOPTransformConfiguration class
+        logger.debug(f"Received configuration: {config}")
+        super().__init__(config)
+        self.seed_urls = config.get("urls", [])
+        self.depth = config.get("depth", 1)
+        self.downloads = config.get("downloads", 10)
+        self.allow_mime_types = config.get("mime_types", ["application/pdf","text/html","text/markdown","text/plain"])
+        self.folder=config.get('folder', None)
+        assert self.seed_urls, "Must specify a list of URLs to crawl. Url cannot be None"
+
+        ## users may be tempted to provide a single URLs, we still need to put it in a list of 1
+        if type(self.seed_urls) is not list:
+            self.seed_urls=[self.seed_urls]
+
+        self.count = 0
+        self.docs = []
+
+    def on_download(self, url: str, body: bytes, headers: dict) -> None:
+        """
+        Callback function called when a page has been downloaded.
+        You have access to the request URL, response body and headers.
+        """
+        doc={}
+        doc['url'] = url
+#        doc['file_size'] = int(headers.get('Content-Length', 0))  # Default to 0 if not found  
+        doc['content_type']=headers.get('Content-Type')
+        try:
+            filename = headers.get('Content-Disposition').split('filename=')[1].strip().strip('"')
+        except:
+            url_split=url.split('/')
+            filename = url_split[-1] if not url.endswith('/') else url_split[-2]
+            filename = filename.replace('.','_')+"-"+doc['content_type'].split(';')[0].replace("/", ".")
+        doc['filename']=filename
+        doc['contents'] = body
+        
+        logger.debug(f"url: {doc['url']}, filename: {doc['filename']}, content_type: {doc['content_type']}")
+
+        ## Enforce download limits
+        if len(self.docs) < self.downloads:
+            self.docs.append(doc)
+
+
+    def transform(self, table: pa.Table=None, file_name: str = None) -> tuple[list[pa.Table], dict[str, Any]]:
+        """
+        Put Transform-specific to convert one Table to 0 or more tables. It also returns
+        a dictionary of execution statistics - arbitrary dictionary
+        This implementation makes no modifications so effectively implements a copy of the
+        input parquet to the output folder, without modification.
+        """
+        start_time = time.time()
+        crawl(
+            self.seed_urls,
+            self.on_download,
+            user_agent=user_agent,
+            depth_limit=self.depth,
+            download_limit=self.downloads,
+            allow_mime_types=self.allow_mime_types
+        )  # blocking call
+
+        # Shutdown all crawls
+        # Check with @Matsubara-san as this is preventing us from calling the transfrom method a second time.
+    #    shutdown()
```

Comment:
We don't need to call the `shutdown()` function here. It will be automatically called when the Python process is terminated gracefully.